Trie/TrieProj/Trie.cs

```diff
@@ -0,0 +1,154 @@
+﻿namespace TrieProj;
+
+/// <summary>
+///  Trie, also called prefix tree, data structure used for store associative array,
+///  with keys that usually are strings, by individual chars.
+/// </summary>
+public class Trie
+{
+    private TrieNode Root { get; }
+
+    /// <summary>
+    /// Constructor of <see cref="Trie"/> class instance 
+    /// </summary>
+    public Trie()
+    {
+        Root = new TrieNode();
+    }
+
+    /// <summary>
+    /// Adds a word in instance of <see cref="Trie"/> class by character
+    /// </summary>
+    public void Add(string word)
+    {
+        var current = Root;
+
+        foreach (var ch in word.ToCharArray())
+        {
+            if (current.Children.ContainsKey(ch))
+                current = current.Children[ch];
```

Comment:
А вот фигурные скобки лучше всё-таки поставить
```suggestion
            if (current.Children.ContainsKey(ch))
            {
                current = current.Children[ch];
            }
```